src/mail.sh

```diff
@@ -0,0 +1,170 @@
+include "$KW_LIB_DIR/kwlib.sh"
+include "$KW_LIB_DIR/kwio.sh"
+include "$KW_LIB_DIR/kw_config_loader.sh"
+
+# Hash containing user options
+declare -gA options_values
+
+function kw_mail()
+{
+  # say "enter"
+  parse_mail_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  if [[ "${options_values['SETUP']}" == 1 ]]; then
+    mail_setup
+    exit
+  fi
+
+  cmd="git send-email $opts --to=$recipients -1"
+
+  # TODO: eval "$cmd"
+  echo "$cmd"
+
+  return 0
+}
+
+function mail_setup()
+{
+  local global=''
+
+  if [[ ! -d "$PWD/.git" ]] && [[ $(ask_yN 'Not in a git repository, apply changes globally?') =~ 1 ]]; then
+    global='--global'
+  fi
+
+  # say 'arrived'
+
+  for option in 'SMTP-USER' 'SMTP-ENCRYPTION' 'SMTP-SERVER' 'SMTP-PORT'; do
+    if [[ -n "${options_values[$option]}" ]]; then
+      echo "git config $global ${options_values[$option]}"
+    fi
+  done
```

Comment:
After reading this function, we could retrieve those options from the kw config as well. Now that we have a `.kw` folder, we probably want to do something like `.kw/mail_config` to save local configurations, and we can have something similar in the `$KW_ETC_DIR`. Ofc, we can also update that file via this mail feature.